canonical-web-and-design/maas-ui
# feat(ui): add dynamic storage pool cards to LXD project summary

## Done

- Added dynamic storage pool cards to the LXD project summary if there are 3 or more storage pools

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the project tab of a LXD KVM
- Open `StorageResources.tsx` and import the storage pool factory on line 10
``` tsx
import { podStoragePool } from "testing/factories";
```
- Replace lines 18-20 with:
``` tsx
const sortedPools = Array.from(Array(2)).map(podStoragePool);
```
- Check that the storage meters show up as before
- Change the amount of pools to 3-4 and check that large cards show
- Change the amount of pools to 5-18 and check that medium cards show
- Change the amount of pools to >18 and check that small cards show
- Try other numbers and resizing the screen and check that the cards are always sized appropriately

## Fixes

Fixes #2415 

# Screenshots
## Large
![Screenshot_2021-04-15 LXD project default bolla MAAS](https://user-images.githubusercontent.com/25733845/114821301-b3966300-9e03-11eb-9e5b-50fc0b7e5d6f.png)

## Medium
![Screenshot_2021-04-15 LXD project default bolla MAAS(1)](https://user-images.githubusercontent.com/25733845/114821309-b5f8bd00-9e03-11eb-89ac-01ce8e350326.png)

## Small
![Screenshot_2021-04-15 LXD project default bolla MAAS(2)](https://user-images.githubusercontent.com/25733845/114821316-b7c28080-9e03-11eb-88f3-4382956ed952.png)
